google-cloud-storage/src/main/java/com/google/cloud/storage/ParallelCompositeUploadBlobWriteSessionConfig.java

```diff
@@ -523,6 +520,35 @@ public static PartNamingStrategy prefix(String prefixPattern) {
       return new WithPrefix(rand, prefixPattern);
     }
 
+    /**
+     * Strategy in which the end object name is the prefix included and is present on each part and
+     * intermediary compose object.
+     *
+     * <p>General format is
+     *
+     * <pre><code>
+     *   {objectName}-parts/{randomKeyDigest};{objectInfoDigest};{partIndex}.part
```

Comment:
I really think this should be updated to not use a directory structure from possible unknowns.